src/context.rs

```diff
@@ -155,6 +156,23 @@ impl Context {
         self.globals.insert(name.to_owned(), val)
     }
 
+    /// Translates a Token to a Value, looking it up in the context if
+    /// necessary
+    ///
+    /// # Panics
+    ///
+    /// Panics if an unsupported token type is passed via t. Only
+    /// Identifiers and Value Literals are supported.
+    pub fn evaluate(&self, t: &Token) -> Option<Value> {
+        match t {
+            &NumberLiteral(f) => Some(Value::Num(f)),
+            &StringLiteral(ref s) => Some(Value::Str(s.clone())),
+            &BooleanLiteral(b) => Some(Value::Bool(b)),
+            &Identifier(ref id) => self.get_val(id).map(|v| v.clone()),
+            _ => panic!("Unsupported token: {:?}", t)
```

Comment:
I think evaluate is a good idea, but we really shouldn't have this panic since it's used on arbitrary user data and we want to guarantee panic-freeness to programmers using the library. If we have to cover the "unsupported token"  case then we should return a Result instead.